eclipse-collections-api/src/main/java/org/eclipse/collections/api/collection/ImmutableCollection.java

```diff
@@ -147,6 +148,15 @@
     @Override
     <V> ImmutableObjectDoubleMap<V> sumByDouble(Function<? super T, ? extends V> groupBy, DoubleFunction<? super T> function);
 
+    /**
+     * @since 8.2
+     */
+    @Override
+    default <V> ImmutableBagIterable<V> countBy(Function<? super T, ? extends V> function)
+    {
+        return this.asLazy().<V>collect(function).toBag().toImmutable();
```

Comment:
In order for the code to be acceptable for the 8.2 release I believe we will need the default implementation.